flink-cdc-runtime/src/main/java/org/apache/flink/streaming/runtime/operators/sink/DataSinkWriterOperator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.operators.sink;
+
+import org.apache.flink.api.common.operators.MailboxExecutor;
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.streaming.api.graph.StreamConfig;
+import org.apache.flink.streaming.api.operators.Output;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.tasks.ProcessingTimeService;
+import org.apache.flink.streaming.runtime.tasks.StreamTask;
+
+import com.ververica.cdc.common.event.Event;
+import com.ververica.cdc.runtime.operators.schema.event.FlushEvent;
+import com.ververica.cdc.runtime.operators.sink.SchemaEvolutionClient;
+
+import java.lang.reflect.Field;
+
+/**
+ * An operator that processes records to be written into a {@link
+ * org.apache.flink.api.connector.sink2.Sink}. It also has a way to process committables with the
+ * same parallelism or send them downstream to a {@link CommitterOperator} with a different
+ * parallelism.
+ *
+ * <p>The operator is always part of a sink pipeline and is the first operator.
+ *
+ * @param <CommT> the type of the committable (to send to downstream operators)
+ */
+public class DataSinkWriterOperator<CommT> extends SinkWriterOperator<Event, CommT> {
+
+    SchemaEvolutionClient schemaEvolutionClient;
+
+    private SinkWriter copySinkWriter;
+
+    private final OperatorID schemaOperatorID;
+
+    DataSinkWriterOperator(
```

Comment:
What about using `public` for the constructor? This operator is located under package `org.apache.flink.xxx`, but callers of the constructor like composer are located under `com.ververica.cdc.xxx`

Reply:
Done. 
BTW, do you know how to create ProcessingTimeService and MailboxExecutor correctly, I couldn't find a suitable method. Setting them to null would have no effect if these components were only used in [Sink.InitContext](https://github.com/apache/flink/blob/5056b0d5dfec25ef90cd0f0e860816ade90d3869/flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/SinkWriterOperator.java#L244), but I'm not sure what it would be used somewhere.